projects/vdk-core/src/vdk/internal/builtin_plugins/run/cli_run.py

```diff
@@ -143,14 +144,41 @@ def __log_exec_result(self, execution_result: ExecutionResult) -> None:
 
     @staticmethod
     def __log_short_exec_result(execution_result):
+        def extract_relevant_lines(step):
+            out = []
+            if step.exception:
+                call_list = traceback.format_tb(step.exception.__traceback__)
+
+                for i, line in enumerate(call_list):
+                    # Check if the step name is in the line
+                    if step.name in line:
+                        out.append(line)
+                        j = i + 1
+                        # Pull in subsequent relevant lines
+                        # that do not come from another file
+                        while j < len(call_list) and not re.match(
+                            r'^\s*File "', call_list[j]
+                        ):
```

Comment:
How robust is this logic? Is there exception format specification we can trust and refer to. 

I really don't want to put anything in core that may break in different version of python or in some corner cases stack traces.

Reply:
I couldn't find any specific docs that describe the traceback format. There are PEPs related to enhancing the traceback, but that's pretty much it.

We could refer to the module code itself. https://github.com/python/cpython/blob/main/Lib/traceback.py#L52
Or to the interpreter code https://github.com/python/cpython/blob/main/Python/traceback.c#L711

They don't seem to change very often, so we should be ok.